### DISCUSSION TITLE
Example of configuration for Monaco

### DISCUSSION THREAD

**Discussion by Stephane-Pires** [slidevjs/slidev#646] [Q&A] [answered]
First, thanks you for this wonderful tool which is "Slidev" it's still a BETA but it feels really polished. 

---------

Lately,  i have been working on a theme for Slidev.  

I have read the documentation : https://sli.dev/custom/config-monaco.html

I have been trying to configure "monaco" using `defineMonacoSetup` which expose the monaco "API, but i have been unsuccessful at configuring it.  

So i decided  to delve inside the source code of "slidev" .

At: https://github.com/slidevjs/slidev/blob/92db1756d72cacfd34ac39a2052a1f242afe013d/packages/client/iframes/monaco/index.ts

I  found that the monaco instance is already configured with 

```ts
  editor = monaco.editor.create(container, {
    model,
    tabSize: 2,
    insertSpaces: true,
    detectIndentation: false,
    folding: false,
    fontSize: +style.getPropertyValue('--slidev-code-font-size').replace(/px/g, ''),
    fontFamily: style.getPropertyValue('--slidev-code-font-family'),
    lineHeight: +style.getPropertyValue('--slidev-code-line-height').replace(/px/g, ''),
    lineDecorationsWidth: 0,
    lineNumbersMinChars: 0,
    scrollBeyondLastLine: false,
    scrollBeyondLastColumn: 0,
    automaticLayout: true,
    readOnly: props.readonly,
    theme: 'vitesse-dark',
    lineNumbers: props.lineNumbers as any,
    glyphMargin: false,
    scrollbar: {
      useShadows: false,
      vertical: 'hidden',
      horizontal: 'hidden',
    },
    overviewRulerLanes: 0,
    minimap: { enabled: false },
  })
```

I would like to be able to override this configuration.  

Any idea ? my goal at start, was to configure "lineNumbers" to "on" and "readOnly" to `true`

Thanks you in advance, for taking the time to  consider this message

Stéphane Pires

**Comment by antfu** [comment, marked as answer]
I have to admit that the Monaco support is pretty barebone since I have used it for only one talk and later not actively looked into it. So there are probably quite some hard-coded parts that I wrote to satisfy only my needs. I am open to exposing better setup APIs to provide full customizability for configuring it. If you see anything missing for you to achieve your needs, feel free to drop some proposals or even PRs. Thanks!

**Reply by Stephane-Pires** [reply]
>I am open to exposing better setup APIs to provide full customizability for configuring it. If you see anything missing for you to achieve your needs, feel free to drop some proposals or even PRs

Erf, sadly my knowledge of "vue" is too restricted at the moment. I don't think i am able to propose a PR to solve my problem cleanly. 

I will dive in a bit to see if i can propose something

👍
